### PR TITLE
[KnownBits] Make `avg{Ceil,Floor}S` optimal

### PR DESCRIPTION
**Pull request by jayfoad**
Rewrite the signed functions in terms of the unsigned ones which are
already optimal.

### CODE REVIEW

**Comment by jayfoad** [comment]
This is an alternative to #110329. I think it's simpler. I don't know if it's any slower or faster.

File: llvm/lib/Support/KnownBits.cpp [hunk 1]
```diff
@@ -763,35 +763,30 @@ KnownBits KnownBits::usub_sat(const KnownBits &LHS, const KnownBits &RHS) {
...
 KnownBits KnownBits::avgCeilS(const KnownBits &LHS, const KnownBits &RHS) {
-  return avgCompute(LHS, RHS, /* IsCeil */ true,
-                    /* IsSigned */ true);
+  return flipSignBit(avgCeilU(flipSignBit(LHS), flipSignBit(RHS)));
 }
 
 KnownBits KnownBits::avgCeilU(const KnownBits &LHS, const KnownBits &RHS) {
-  return avgCompute(LHS, RHS, /* IsCeil */ true,
-                    /* IsSigned */ false);
+  return avgComputeU(LHS, RHS, /* IsCeil */ true);
```
Comment (goldsteinn) [inline llvm/lib/Support/KnownBits.cpp:789]: Nit, `/*IsCeil=*/` for the comments.

**Review by goldsteinn — APPROVED** [review]
LGTM, this is much cleaner

File: llvm/lib/Support/KnownBits.cpp (hunk 1 above)
Comment (jayfoad) [reply llvm/lib/Support/KnownBits.cpp]: Done.

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**